cristin-person/src/test/java/no/unit/nva/cristin/person/handler/FetchFromIdentityNumberHandlerTest.java

```diff
@@ -0,0 +1,201 @@
+import static no.unit.nva.cristin.person.handler.FetchFromIdentityNumberHandler.NIN_TYPE;
+import static no.unit.nva.utils.AccessUtils.EDIT_OWN_INSTITUTION_USERS;
+import static nva.commons.apigateway.MediaTypes.APPLICATION_PROBLEM_JSON;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class FetchFromIdentityNumberHandlerTest {
+
+    private static final Map<String, String> EMPTY_MAP = Collections.emptyMap();
+    public static final Map<String, String> INVALID_PAYLOAD = Map.of("SomeKey", "SomeValue");
+    private static final String NVA_API_GET_PERSON_RESPONSE_JSON =
+        "nvaApiGetPersonResponse.json";
+    private static final String DEFAULT_IDENTITY_NUMBER = "07117631634";
+    private static final String VALID_CRISTIN_NATIONAL_ID_URI = "https://api.cristin-test.uio"
+        + ".no/v2/persons?national_id=07117631634"; // &lang=en,nb,nn"; No need for language in first fetch
+    private static final String URI_FIRST_HIT_FROM_CRISTIN = "https://api.cristin-test.uio"
+        + ".no/v2/persons/359084?lang=en,nb,nn";
+    public static final Map<String, String> SOME_QUERY_PARAM = Map.of("SomeQueryParam", "SomeValue");
+    public static final String WRONG_TYPE = "WRONGTYPE";
+    public static final String WRONG_VALUE = "wrongValue";
+    public static final String EMPTY_ARRAY = "[]";
+    private static final String EMPTY_STRING = "";
+
+    private final Environment environment = new Environment();
+    private AuthorizedCristinPersonApiClient apiClient;
+    private Context context;
+    private ByteArrayOutputStream output;
+    private FetchFromIdentityNumberHandler handler;
+
+    @BeforeEach
+    void setUp() {
+        apiClient = new AuthorizedCristinPersonApiClientStub();
+
+        context = mock(Context.class);
+        output = new ByteArrayOutputStream();
+        handler = new FetchFromIdentityNumberHandler(apiClient, environment);
+    }
+
+    @Test
+    void shouldReturnPersonWhenMockedHttpResponseContainsCristinPerson() throws Exception {
+        Person actual = sendQuery(defaultBody(), EMPTY_MAP).getBodyObject(Person.class);
+        String expectedString = IoUtils.stringFromResources(Path.of(NVA_API_GET_PERSON_RESPONSE_JSON));
+        Person expected = OBJECT_MAPPER.readValue(expectedString, Person.class);
+
+        assertThat(actual, equalTo(expected));
+    }
+
+    @Test
+    void typedValueTest() throws IOException {
```

Comment:
Is this necessary?

Reply:
Not at all, just to make a valid request at a point